### DISCUSSION TITLE
File to import not found or unreadable: .../bootstrap@v5.3.3+incompatible/scss/_functions.scss.

### DISCUSSION THREAD

**Discussion by FuadEfendi** [hbstack/.github#60] [Issues and bugs]
When I try to build large sites locally, I have this issues:
```
ERROR TOCSS: failed to transform "hb/scss/index.scss" (text/x-scss): "/Users/fefendi/Library/Caches/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/hb@v0.12.2/assets/hb/scss/index.tmpl.scss:4:1": File to import not found or unreadable: /Users/fefendi/Library/Caches/hugo_cache/modules/filecache/modules/pkg/mod/github.com/twbs/bootstrap@v5.3.3+incompatible/scss/_functions.scss.
Built in 1674717 ms
Error: error building site: TOCSS: failed to transform "hb/scss/index.scss" (text/x-scss): "/Users/fefendi/Library/Caches/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/hb@v0.12.2/assets/hb/scss/index.tmpl.scss:4:1": File to import not found or unreadable: /Users/fefendi/Library/Caches/hugo_cache/modules/filecache/modules/pkg/mod/github.com/twbs/bootstrap@v5.3.3+incompatible/scss/_functions.scss.
```

I delete cache, upgrade, etc.; still have this issue. I don't have such issue with small sites (1000 pages); only with large sites (10000 pages); and I don't have such issue with Netlify. I delete all caches, it doesn't help...

**Comment by FuadEfendi** [comment]
I had this issue with `hugo v0.126.1+extended darwin/arm64 BuildDate=2024-05-15T10:42:34Z VendorInfo=brew` - and it was only with huge sites, localy; but same site at Netlify didn't have this issue, and it was using `0.125.5` version at Netlify.

I first upgraded to `0.127.0` (released few days ago) - same issue.

Ten downgraded to `v0.125.5-c8b9f9f81c375f5b391e61bae711ee63fc76c1fd darwin/arm64 BuildDate=2024-05-01T15:22:11Z VendorInfo=gohugoio` which I downloaded from https://github.com/gohugoio/hugo/releases, unpacked, and used without any "installation", and it works, no any issues.

I noticed some old post mentioning absolute vs. relative path, with similar error.

**Comment by razonyang** [comment]
I couldn't reproduce this with Hugo `v0.126.3` and `0.127.0`, could you please provide a minimal repo for reproduction.

![image](https://github.com/hbstack/.github/assets/17720932/582e49be-6e5c-49ba-9260-8b47aa81e731)
![image](https://github.com/hbstack/.github/assets/17720932/fe865a56-c761-4858-ac30-a8d86fb3dbab)

I believe this issue was fixed a long time ago.

**Comment by razonyang** [comment]
Also works on 0.126.1
![image](https://github.com/hbstack/.github/assets/17720932/489c892f-c84a-43a7-abeb-5667e8540eec)

**Comment by razonyang** [comment]
> I don't have such issue with small sites (1000 pages); only with large sites (10000 pages)

Just a guess, would this take too much resources (such as disk spaces) on you laptop, and causes this?

Btw, you can use [cascade](https://gohugo.io/getting-started/configuration/#cascade) to ignore some content if you just want to preview the site partially on local.

**Reply by FuadEfendi** [reply]
From my experiments, it happens with huge repositories only, but it is consistent (always happens with the same repository); maybe some of documents causes that, or **multithreading** issues. Never happens at Netlify (0.125.5)

Yes, on laptop during development I started to use `hugo server --renderToMemory` command which improved build time significantly (otherwise my SSD can burn...); I also wrote Java code which restarts server, catches faulty file (file with malformed markdown preventing Hugo to start), moves it into another folder, logs message into file, and restarts, repeats it in a loop until 100-200 faulty files removed... unfortunately Hugo can report single file only.

**Reply by razonyang** [reply]
Have no idea on this, it not likely a theme issue, and hard to reproduce and locate the cause, will try if you can provide the source of your repo, but please don't expect too much.

**Reply by FuadEfendi** [reply]
Thanks; when I created it, I thought it is bug, but now I feel it is environment specific, please close it.
